core/src/main/scala/dev/chronicles/core/VersionTracker.scala

```diff
@@ -60,14 +68,19 @@ trait VersionTracker[F[_]] {
     */
   def setCurrentVersion(table: TableName, id: VersionTracker.CommitId): F[Unit]
 
+  /**
+    * Get the flag that indicates whether a table is a snapshot table or not.
+    */
+  def isSnapshotTable(table: TableName): F[Boolean]
+
   //
   // Internal operations to be provided by implementations
   //
 
   /**
     * Produce description of the current state of table.
     */
-  protected def tableState(table: TableName): F[TableState]
+  protected def tableState(table: TableName, timeOrder: Boolean): F[TableState[F]]
```

Comment:
A custom sum might be nice here to avoid the boolean blindness:

```scala
object TableState {
  sealed trait Ordering
  object Ordering {
    final case object RecentFirst extends Ordering
    final case object RecentLast extends Ordering
  }
}
```

You could also look to retain the ordering in the types with something like:

```scala
protected def tableState[O <: TableState.Ordering](table: TableName, ordering: O): F[TableState[F, O]]
```

Then you could make it a compilation error to derive the current version from anything but a `TableState[F, RecentLast]`.

Probs overkill for now, but perhaps useful if this ordering constraint shows up more in the future.

Reply:
> A custom sum might be nice here to avoid the boolean blindness

Ha yes I thought that when doing this, I was just being lazy... 😄

> You could also look to retain the ordering in the types with something like:

I like the idea for encoding the ordering in the type very much! These are the sort of gold nuggets that make your reviews so useful, Tom! 😊👍

Reply:
I did add the order type as a parameter to `TableState`, I don't actually think it would catch any errors the way things work now, but it seems like a nice way to describe what it does anyway.